src/main/java/com/amuzil/omegasource/magus/skill/conditionals/key/KeyPermutation.java

```diff
@@ -9,7 +12,39 @@
  * ONLY SUPPORTS MULTIPLE KEYS AT ONCE. DELAY IS USELESS HERE. If multiple keys are passed with *different* delay,
  * the key with the most delay will be read.
  */
-public record KeyPermutation (
-        LinkedList<KeyInput> keys
-) {
+public final class KeyPermutation implements InputData {
```

Comment:
Why make this not a record? You can still override methods, instead you've made extra work for yourself writing the getters and setters